### PR TITLE
reported issue with proposed fix that breaks tests

### PR DESCRIPTION
**Pull request by fil512**
Reported in Issue #1241

### CODE REVIEW

**Comment by jamesagnew** [comment]
@fil512 Heh, it won't let me ask you for a review because you originally opened this PR to add the tests.

Uhhhhh can you review? :)

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirSystemDaoR4Test.java [hunk 1]
```diff
@@ -2429,6 +2385,66 @@ public void testTransactionWithRefsToConditionalCreate() {
 
 	}
 
+
+	@Test
+	public void testDeleteInTransactionShouldFailWhenReferencesExist() {
+		final Observation obs1 = new Observation();
+		obs1.setStatus(ObservationStatus.FINAL);
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirSystemDaoR4Test.java]: Why are you setting the status to FINAL?

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/TransactionDeleteR4Test.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+		myObservationDao.read(obs1id);
+		myObservationDao.read(obs2id);
+
+		rpt = new DiagnosticReport();
+		rpt.addIdentifier().setSystem("foo").setValue("IDENTIFIER");
+		rpt.addResult(new Reference(obs2id));
+
+		Bundle b = new Bundle();
+		b.addEntry().getRequest().setMethod(Bundle.HTTPVerb.DELETE).setUrl("Observation?_has:DiagnosticReport:result:identifier=foo|IDENTIFIER");
```
Comment (fil512) [inline hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/TransactionDeleteR4Test.java]: This is an interesting edge case to test.  But it feels like you're not testing the more straightforward case:
create obs1, obs2, rpt with obs2

alter rpt object, removing reference to obs2 (or changing reference to obs1 if you want to get fancy)

bundle {
delete obj2
update rpt
}

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirSystemDaoR4Test.java (hunk 1 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/FhirSystemDaoR4Test.java]: final isn't relevant, just setting a property so that the resource has some content

File: hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/TransactionDeleteR4Test.java (hunk 2 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/TransactionDeleteR4Test.java]: Do you not feel that `testDeleteWithHas_TargetModifiedToNoLongerIncludeReference` is testing that scenario?

Comment (fil512) [reply hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/TransactionDeleteR4Test.java]: It PROBABLY does.  But there are two differences:
1) In that test, the report doesn't exist before the transaction.
2) The delete for the observation is by a query which could return different results inside and outside the transaction, whereas delete by id is more definitive.

Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/test/java/ca/uhn/fhir/jpa/dao/r4/TransactionDeleteR4Test.java]: Got it. Ok, one more test added.
